### PR TITLE
Don't generate group attestations for local users

### PR DESCRIPTION
**Pull request by erikjohnston**
This should help with https://github.com/matrix-org/synapse/issues/2592

### CODE REVIEW

**Comment by richvdh** [comment]
@erikjohnston: tests are failing

File: synapse/storage/group_server.py [hunk 1]
```diff
@@ -1086,6 +1086,24 @@ def update_remote_attestion(self, group_id, user_id, attestation):
...
+        """Remove an attestation that we thought we should renew, but actually
+        shouldn't. Ideally this would never get called as we would never
+        incorrectly try and do attestations for local users on local groups.
+
+        Args:
+            group_id (str)
+            user_id (str)
+        """
+        return self._simple_update_one(
+            table="_simple_delete",
```
Comment (richvdh) [inline synapse/storage/group_server.py]: this looks bogus

File: synapse/groups/attestations.py [hunk 2]
```diff
@@ -130,10 +130,17 @@ def _renew_attestations(self):
         def _renew_attestation(group_id, user_id):
             attestation = self.attestations.create_attestation(group_id, user_id)
 
-            if self.is_mine_id(group_id):
+            if not self.is_mine_id(group_id):
+                destination = get_domain_from_id(group_id)
+            else not self.is_mine_id(user_id):
```
Comment (richvdh) [inline synapse/groups/attestations.py]: else not

File: synapse/groups/attestations.py [hunk 3]
```diff
@@ -130,10 +130,17 @@ def _renew_attestations(self):
         def _renew_attestation(group_id, user_id):
             attestation = self.attestations.create_attestation(group_id, user_id)
```
Comment (richvdh) [inline synapse/groups/attestations.py]: could move this to after the conditions? <shrug>

**Comment by erikjohnston** [comment]
> My main question is how we've managed to get so far without this being a problem. Is it just that it works but is inefficient so was fine until it landed on a heavily-loaded server?

Yes, plus that we can't really do sytests for the renewals so I need to figure out how to test it in unit tests.

**Pull request merged**